Bug: foundry fails building due to mismatched types

### Component

Other (please describe)

### Have you ensured that all of these are up to date?

- [ ] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

_No response_

### What command(s) is the bug in?

cargo build --bins --release

### Operating System

_No response_

### Describe the bug

```
error[E0308]: mismatched types
  --> cli/src/cmd/forge/watch.rs:95:21
   |
95 |     runtime.command(cmd.clone());
   |                     ^^^^^^^^^^^ expected enum `watchexec::command::Command`, found struct `Vec`
   |
   = note: expected enum `watchexec::command::Command`
            found struct `Vec<std::string::String>`

error[E0061]: this function takes 2 arguments but 1 argument was supplied
   --> cli/src/cmd/forge/watch.rs:101:8
    |
101 |     wx.send_event(Event::default()).await?;
    |        ^^^^^^^^^^ ---------------- supplied 1 argument
    |        |
    |        expected 2 arguments
    |
note: associated function defined here
   --> /opt/foundry/.cargo/registry/src/github.com-1ecc6299db9ec823/watchexec-2.0.0/src/watchexec.rs:156:15
    |
156 |     pub async fn send_event(&self, event: Event, priority: Priority) -> Result<(), CriticalError> {
    |                  ^^^^^^^^^^

error[E0308]: mismatched types
   --> cli/src/cmd/forge/watch.rs:114:21
    |
114 |     runtime.command(cmd.clone());
    |                     ^^^^^^^^^^^ expected enum `watchexec::command::Command`, found struct `Vec`
    |
    = note: expected enum `watchexec::command::Command`
             found struct `Vec<std::string::String>`

error[E0061]: this function takes 2 arguments but 1 argument was supplied
   --> cli/src/cmd/forge/watch.rs:120:8
    |
120 |     wx.send_event(Event::default()).await?;
    |        ^^^^^^^^^^ ---------------- supplied 1 argument
    |        |
    |        expected 2 arguments
    |
note: associated function defined here
   --> /opt/foundry/.cargo/registry/src/github.com-1ecc6299db9ec823/watchexec-2.0.0/src/watchexec.rs:156:15
    |
156 |     pub async fn send_event(&self, event: Event, priority: Priority) -> Result<(), CriticalError> {
    |                  ^^^^^^^^^^

error[E0308]: mismatched types
   --> cli/src/cmd/forge/watch.rs:132:21
    |
132 |     runtime.command(cmd.clone());
    |                     ^^^^^^^^^^^ expected enum `watchexec::command::Command`, found struct `Vec`
    |
    = note: expected enum `watchexec::command::Command`
             found struct `Vec<std::string::String>`

error[E0061]: this function takes 2 arguments but 1 argument was supplied
   --> cli/src/cmd/forge/watch.rs:154:8
    |
154 |     wx.send_event(Event::default()).await?;
    |        ^^^^^^^^^^ ---------------- supplied 1 argument
    |        |
    |        expected 2 arguments
    |
note: associated function defined here
   --> /opt/foundry/.cargo/registry/src/github.com-1ecc6299db9ec823/watchexec-2.0.0/src/watchexec.rs:156:15
    |
156 |     pub async fn send_event(&self, event: Event, priority: Priority) -> Result<(), CriticalError> {
    |                  ^^^^^^^^^^

error[E0308]: mismatched types
   --> cli/src/cmd/forge/watch.rs:206:24
    |
206 |         config.command(cmd.clone());
    |                        ^^^^^^^^^^^ expected enum `watchexec::command::Command`, found struct `Vec`
    |
    = note: expected enum `watchexec::command::Command`
             found struct `Vec<std::string::String>`

error[E0308]: mismatched types
   --> cli/src/cmd/forge/watch.rs:245:20
    |
245 |     config.command(new_cmd);
    |                    ^^^^^^^ expected enum `watchexec::command::Command`, found struct `Vec`
    |
    = note: expected enum `watchexec::command::Command`
             found struct `Vec<std::string::String>`

error[E0599]: no method named `command_shell` found for struct `RuntimeConfig` in the current scope
   --> cli/src/cmd/forge/watch.rs:396:12
    |
396 |     config.command_shell(default_shell());
    |            ^^^^^^^^^^^^^ method not found in `RuntimeConfig`

error[E0599]: no variant or associated item named `default` found for enum `watchexec::command::Shell` in the current scope
   --> cli/src/cmd/forge/watch.rs:419:12
    |
419 |     Shell::default()
    |            ^^^^^^^ variant or associated item not found in `watchexec::command::Shell`

Some errors have detailed explanations: E0061, E0308, E0599.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `foundry-cli` due to 10 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
```

running `cargo build --bins --release`  

using rustc 1.59,
debian bullseye

--- a/cli/Cargo.toml
+++ b/cli/Cargo.toml
@@ -41,7 +41,7 @@ tracing-error = "0.2.0"
 tracing-subscriber = { version = "0.3", features = ["registry", "env-filter", "fmt"] }
 tracing = "0.1.26"
 console = "0.15.0"
-watchexec = "2.0.0-pre.14"
+watchexec = "2.0.0"
 atty = "0.2.14"
 comfy-table = "5.0.0"
 

--- a/cli/src/cmd/forge/watch.rs
+++ b/cli/src/cmd/forge/watch.rs
@@ -5,15 +5,14 @@ use crate::{
     utils::{self, FoundryPathExt},
 };
 use clap::Parser;
-
 use foundry_config::Config;
 use std::{collections::HashSet, convert::Infallible, path::PathBuf, sync::Arc};
 use tracing::trace;
 use watchexec::{
     action::{Action, Outcome, PreSpawn},
-    command::Shell,
+    command::Command,
     config::{InitConfig, RuntimeConfig},
-    event::{Event, ProcessEnd},
+    event::{Event, Priority, ProcessEnd},
     handler::SyncFnHandler,
     paths::summarise_events_to_env,
     signal::source::MainSignal,
@@ -92,13 +91,13 @@ pub async fn watch_build(args: BuildArgs) -> eyre::Result<()> {
     let cmd = cmd_args(args.watch.watch.as_ref().map(|paths| paths.len()).unwrap_or_default());
 
     trace!("watch build cmd={:?}", cmd);
-    runtime.command(cmd.clone());
+    runtime.command(watch_command(cmd.clone()));
 
     let wx = Watchexec::new(init, runtime.clone())?;
     on_action(args.watch, runtime, Arc::clone(&wx), cmd, (), |_| {});
 
     // start executing the command immediately
-    wx.send_event(Event::default()).await?;
+    wx.send_event(Event::default(), Priority::default()).await?;
     wx.main().await??;
 
     Ok(())
@@ -111,13 +110,13 @@ pub async fn watch_snapshot(args: SnapshotArgs) -> eyre::Result<()> {
     let cmd = cmd_args(args.test.watch.watch.as_ref().map(|paths| paths.len()).unwrap_or_default());
 
     trace!("watch snapshot cmd={:?}", cmd);
-    runtime.command(cmd.clone());
+    runtime.command(watch_command(cmd.clone()));
     let wx = Watchexec::new(init, runtime.clone())?;
 
     on_action(args.test.watch.clone(), runtime, Arc::clone(&wx), cmd, (), |_| {});
 
     // start executing the command immediately
-    wx.send_event(Event::default()).await?;
+    wx.send_event(Event::default(), Priority::default()).await?;
     wx.main().await??;
 
     Ok(())
@@ -129,7 +128,7 @@ pub async fn watch_test(args: TestArgs) -> eyre::Result<()> {
     let (init, mut runtime) = args.watchexec_config()?;
     let cmd = cmd_args(args.watch.watch.as_ref().map(|paths| paths.len()).unwrap_or_default());
     trace!("watch test cmd={:?}", cmd);
-    runtime.command(cmd.clone());
+    runtime.command(watch_command(cmd.clone()));
     let wx = Watchexec::new(init, runtime.clone())?;
 
     let config: Config = args.build_args().into();
@@ -151,7 +150,7 @@ pub async fn watch_test(args: TestArgs) -> eyre::Result<()> {
     on_action(args.watch.clone(), runtime, Arc::clone(&wx), cmd, state, on_test);
 
     // start executing the command immediately
-    wx.send_event(Event::default()).await?;
+    wx.send_event(Event::default(), Priority::default()).await?;
     wx.main().await??;
 
     Ok(())
@@ -203,7 +202,7 @@ fn on_test(action: OnActionState<WatchTestState>) {
         // run, or if no test files were changed and no previous test files were modified in which
         // case we simply run all
         let mut config = runtime.clone();
-        config.command(cmd.clone());
+        config.command(watch_command(cmd.clone()));
         // re-register the action
         on_action(
             args.clone(),
@@ -242,7 +241,7 @@ fn on_test(action: OnActionState<WatchTestState>) {
 
     // reconfigure the executor with a new runtime
     let mut config = runtime.clone();
-    config.command(new_cmd);
+    config.command(watch_command(new_cmd));
 
     // re-register the action
     on_action(
@@ -255,6 +254,18 @@ fn on_test(action: OnActionState<WatchTestState>) {
     );
 }
 
+/// Converts a list of arguments to a `watchexec::Command`
+///
+/// The first index in `args`, is expected to be the path to the executable, See `cmd_args`
+///
+/// # Panics
+/// if `args` is empty
+fn watch_command(mut args: Vec<String>) -> Command {
+    debug_assert!(!args.is_empty());
+    let prog = args.remove(0);
+    Command::Exec { prog, args }
+}
+
 /// Returns the env args without the `--watch` flag from the args for the Watchexec command
 fn cmd_args(num: usize) -> Vec<String> {
     // all the forge arguments including path to forge bin
@@ -262,6 +273,7 @@ fn cmd_args(num: usize) -> Vec<String> {
     if let Some(pos) = cmd_args.iter().position(|arg| arg == "--watch" || arg == "-w") {
         cmd_args.drain(pos..=(pos + num));
     }
+
     cmd_args
 }
 
@@ -269,7 +281,7 @@ fn cmd_args(num: usize) -> Vec<String> {
 pub fn init() -> eyre::Result<InitConfig> {
     let mut config = InitConfig::default();
     config.on_error(SyncFnHandler::from(|data| -> std::result::Result<(), Infallible> {
-        tracing::trace!("[[{:?}]]", data);
+        trace!("[[{:?}]]", data);
         Ok(())
     }));
 
@@ -393,8 +405,6 @@ pub fn runtime(args: &WatchArgs) -> eyre::Result<RuntimeConfig> {
         config.action_throttle(utils::parse_delay(delay)?);
     }
 
-    config.command_shell(default_shell());
-
     config.on_pre_spawn(move |prespawn: PreSpawn| async move {
         let envs = summarise_events_to_env(prespawn.events.iter());
         if let Some(mut command) = prespawn.command().await {
@@ -407,14 +417,4 @@ pub fn runtime(args: &WatchArgs) -> eyre::Result<RuntimeConfig> {
     });
 
     Ok(config)
-}
-
-#[cfg(windows)]
-fn default_shell() -> Shell {
-    Shell::Powershell
-}
-
-#[cfg(not(windows))]
-fn default_shell() -> Shell {
-    Shell::default()
 }

